tests/e2e/support/page/files/publicLink.ts

```diff
@@ -0,0 +1,228 @@
+    this.roleDropdownLocator = this.actor.page.locator('#files-file-link-role-button')
+    this.expirationDateDropdownLocator = this.actor.page.locator('#files-links-expiration-btn')
+    this.createLinkButtonLocator = this.actor.page.locator('#oc-files-file-link-create')
+    this.publicLinkLocator = this.actor.page.locator(
+      '//a[@class = "oc-files-file-link-url oc-text-truncate"]'
+    )
+    this.yearButtonLocator = this.actor.page.locator(
+      `//div[@class = "vc-nav-container"]/div[@class="vc-nav-header"]//span[position()=2]`
+    )
+    this.nextSpanYearLocator = this.actor.page.locator(
+      `//div[@class = "vc-nav-container"]/div[@class="vc-nav-header"]//span[position()=3]`
+    )
+    this.monthAndYearDropdownLocator = this.actor.page.locator(`//div[@class = 'vc-title']`)
+  }
+
+  async selectRole(role: string): Promise<void> {
+    await this.roleDropdownLocator.click()
+    await this.actor.page.locator(`//span[@id="files-role-${role}"]`).click()
+  }
+
+  async selectExpiryMonth(year: string, month: string): Promise<void> {
+    const monthLocator = await this.actor.page.locator(`//span[@data-id='${year}.${month}']`)
+    await monthLocator.click()
+  }
+
+  async selectExpiryDay(dayMonthYear: string): Promise<void> {
+    const dayLocator = await this.actor.page.locator(
+      `//span[@tabindex='-1' or @tabindex='0'][@aria-label='${dayMonthYear}']`
+    )
+    await dayLocator.click()
+  }
+
+  checkDaysType = (stringDate: string): string => {
+    if (stringDate.includes('years')) {
+      return 'year'
+    } else if (stringDate.includes('months')) {
+      return 'month'
+    } else if (stringDate.includes('weeks')) {
+      return 'week'
+    } else {
+      return 'day'
+    }
+  }
+
+  isCheckDate = (expiryDate: string): string => {
+    if (expiryDate.charAt(0).includes('-')) {
+      throw new Error('The provided date is negative and has already expired !!')
+    } else if (expiryDate.charAt(0).includes('+')) {
+      return this.checkDaysType(expiryDate)
+    }
+  }
+
+  addMonth = (noOfMonths: number): Date => {
+    const date = new Date()
+    date.setMonth(date.getMonth() + noOfMonths)
+    return date
+  }
+
+  addDays = (date: Date, days: number): Date => {
+    date.setDate(date.getDate() + days)
+    return date
+  }
+
+  setActualExpiryDate = (dateType: string, dataOfExpiration: string): Date => {
+    switch (dateType) {
+      case 'day':
+        return this.addDays(new Date(), parseInt(dataOfExpiration))
+      case 'week':
+        return this.addDays(new Date(), parseInt(dataOfExpiration) * 7)
+      case 'month':
+        return this.addMonth(parseInt(dataOfExpiration))
+      case 'year':
+        return new Date(
+          new Date().setFullYear(new Date().getFullYear() + parseInt(dataOfExpiration))
+        )
+    }
+  }
+
+  async selectDate(dataOfExpiration: string): Promise<void> {
+    await this.actor.page.pause()
```

Comment:
Why pause?

Reply:
@SwikritiT sorry my bad forgot to remove it.